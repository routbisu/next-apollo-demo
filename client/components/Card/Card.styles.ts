import styled, { css } from 'styled-components'
import { colors } from '../../styling/tokens'

type CardProps = {
  isLoading?: boolean
}

const loadingStyles = css`
  @keyframes example {
    0% {
      opacity: 0.5;
    }
    50% {
      opacity: 1;
    }
    100% {
      opacity: 0.5;
    }
  }

  animation-name: example;
  animation-duration: 0.4s;
  animation-iteration-count: infinite;
`

export const CardWrapper = styled('div')<CardProps>`
  padding: 8px;
  display: flex;
  flex-direction: column;

  ${({ isLoading }) => isLoading && loadingStyles}
  min-height: 175px;
`

export const CardContent = styled('div')`
  padding: 16px;
  background: ${colors.gray[100]};
  border-radius: 8px;
  flex: 1;
`

export const UserName = styled('h2')`
  color: ${colors.orange[500]};
  font-weight: 600;
  font-size: 1.1rem;
  margin-bottom: 16px;
  line-height: 1.25rem;
`

export const UserPhone = styled('a')`
  display: inline-block;
  font-size: 0.95rem;
  font-weight: 500;
  margin-bottom: 8px;
  color: ${colors.blue[500]};
  text-decoration: none;
`

export const UserEmail = styled('a')`
  font-size: 0.85rem;
  display: inline-block;
  margin-bottom: 8px;
  font-weight: 400;
  color: ${colors.blue[500]};
  text-decoration: none;
`

export const UserAddress = styled('p')`
  font-size: 0.85rem;
  color: ${colors.gray[500]};
  line-height: 1.25rem;
`
