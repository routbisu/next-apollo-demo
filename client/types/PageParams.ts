export type PageParams = {
  pageNumber?: number
  pageSize?: number
}
