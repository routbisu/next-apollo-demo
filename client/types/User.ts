export type User = {
  id?: string
  name?: string
  email?: string
  address?: string
  phone?: string
}
