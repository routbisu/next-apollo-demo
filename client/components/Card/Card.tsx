import React from 'react'
import { User } from '../../types/User'
import {
  CardContent,
  CardWrapper,
  UserAddress,
  UserEmail,
  UserName,
  UserPhone
} from './Card.styles'

type CardProps = Omit<User, 'id'> & {
  isLoading?: boolean
}

const Card: React.FC<CardProps> = ({
  name,
  phone,
  email,
  address,
  ...rest
}) => {
  return (
    <CardWrapper {...rest}>
      <CardContent>
        <UserName data-testid="name">{name}</UserName>
        <div>
          <UserPhone data-testid="phone" href={`tel:${phone}`}>
            {phone}
          </UserPhone>
        </div>
        <div>
          <UserEmail data-testid="email" href={`mailto:${email}`}>
            {email}
          </UserEmail>{' '}
        </div>
        <UserAddress data-testid="address">{address}</UserAddress>
      </CardContent>
    </CardWrapper>
  )
}

export default Card
