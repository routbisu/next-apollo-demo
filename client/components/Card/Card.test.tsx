/**
 * @jest-environment jsdom
 */

import React from 'react'
import { shallow } from 'enzyme'
import Card from './Card'
import { User } from 'types/User'

describe('card component', () => {
  it('should render the card correctly', async () => {
    const props: User = {
      name: 'James Bond',
      phone: '1234567890',
      email: 'james.bond@gmail.com',
      address: '30 Wellington Square, Chelsea'
    }

    const wrapper = shallow(<Card {...props} />)

    // Snapshot test
    expect(wrapper).toMatchSnapshot()

    // Check name
    expect(wrapper.find({ 'data-testid': 'name' }).text()).toEqual('James Bond')
    // Check phone
    expect(wrapper.find({ 'data-testid': 'phone' }).text()).toEqual(
      '1234567890'
    )
    // Check email
    expect(wrapper.find({ 'data-testid': 'email' }).text()).toEqual(
      'james.bond@gmail.com'
    )
    // Check address
    expect(wrapper.find({ 'data-testid': 'address' }).text()).toEqual(
      '30 Wellington Square, Chelsea'
    )
  })
})
