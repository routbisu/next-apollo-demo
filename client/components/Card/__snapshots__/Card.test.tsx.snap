// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card component should render the card correctly 1`] = `
<Cardstyles__CardWrapper>
  <Cardstyles__CardContent>
    <Cardstyles__UserName
      data-testid="name"
    >
      James Bond
    </Cardstyles__UserName>
    <div>
      <Cardstyles__UserPhone
        data-testid="phone"
        href="tel:1234567890"
      >
        1234567890
      </Cardstyles__UserPhone>
    </div>
    <div>
      <Cardstyles__UserEmail
        data-testid="email"
        href="mailto:james.bond@gmail.com"
      >
        james.bond@gmail.com
      </Cardstyles__UserEmail>
       
    </div>
    <Cardstyles__UserAddress
      data-testid="address"
    >
      30 Wellington Square, Chelsea
    </Cardstyles__UserAddress>
  </Cardstyles__CardContent>
</Cardstyles__CardWrapper>
`;
